{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pyranges as pr\n",
    "from pyranges import PyRanges\n",
    "import pandas as pd\n",
    "from io import StringIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# WGS/WES\n",
    "## PATH = /data/share/ICGC/STAD/all_snv_indel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "refGene = pd.read_csv(\"/home/tsubosaka/stomach_normal_marker/refGene.txt.gz\", header=None, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames = [\"number\", \"Name of Gene\", \"Chromosome\", \"Strand\",\n",
    "           \"Start\",#\"Transcription start position\",\n",
    "           \"End\",#\"Transcription end position\",\n",
    "           \"Coding region start\",\n",
    "           \"Coding region end\",\n",
    "           \"Number of exons\",\n",
    "           \"Exon start position\",\n",
    "           \"Exon end position\",\n",
    "           \"Score\",\n",
    "           \"Alternative name\",\n",
    "           \"Status of CDS start annotations(none, unnk, incmpl, cmpl)\",\n",
    "           \"Status of CDS end annotations(none, unnk, incmpl, cmpl)\",\n",
    "           \"exon frame offsets\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "refGene.columns = colnames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>number</th>\n",
       "      <th>Name of Gene</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Coding region start</th>\n",
       "      <th>Coding region end</th>\n",
       "      <th>Number of exons</th>\n",
       "      <th>Exon start position</th>\n",
       "      <th>Exon end position</th>\n",
       "      <th>Score</th>\n",
       "      <th>Alternative name</th>\n",
       "      <th>Status of CDS start annotations(none, unnk, incmpl, cmpl)</th>\n",
       "      <th>Status of CDS end annotations(none, unnk, incmpl, cmpl)</th>\n",
       "      <th>exon frame offsets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>585</td>\n",
       "      <td>NR_148357</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>11868</td>\n",
       "      <td>14362</td>\n",
       "      <td>14362</td>\n",
       "      <td>14362</td>\n",
       "      <td>3</td>\n",
       "      <td>11868,12612,13220,</td>\n",
       "      <td>12227,12721,14362,</td>\n",
       "      <td>0</td>\n",
       "      <td>LOC102725121</td>\n",
       "      <td>unk</td>\n",
       "      <td>unk</td>\n",
       "      <td>-1,-1,-1,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>585</td>\n",
       "      <td>NR_046018</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>11873</td>\n",
       "      <td>14409</td>\n",
       "      <td>14409</td>\n",
       "      <td>14409</td>\n",
       "      <td>3</td>\n",
       "      <td>11873,12612,13220,</td>\n",
       "      <td>12227,12721,14409,</td>\n",
       "      <td>0</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>unk</td>\n",
       "      <td>unk</td>\n",
       "      <td>-1,-1,-1,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2127</td>\n",
       "      <td>NM_139163</td>\n",
       "      <td>chr2</td>\n",
       "      <td>-</td>\n",
       "      <td>202152993</td>\n",
       "      <td>202222121</td>\n",
       "      <td>202153377</td>\n",
       "      <td>202216127</td>\n",
       "      <td>15</td>\n",
       "      <td>202152993,202154179,202154418,202163960,202172...</td>\n",
       "      <td>202153504,202154289,202154577,202164023,202172...</td>\n",
       "      <td>0</td>\n",
       "      <td>FLACC1</td>\n",
       "      <td>cmpl</td>\n",
       "      <td>cmpl</td>\n",
       "      <td>2,0,0,0,1,0,1,2,0,2,0,2,2,0,-1,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>585</td>\n",
       "      <td>NR_106918</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>17368</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>1</td>\n",
       "      <td>17368,</td>\n",
       "      <td>17436,</td>\n",
       "      <td>0</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>unk</td>\n",
       "      <td>unk</td>\n",
       "      <td>-1,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>585</td>\n",
       "      <td>NR_107062</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>17368</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>1</td>\n",
       "      <td>17368,</td>\n",
       "      <td>17436,</td>\n",
       "      <td>0</td>\n",
       "      <td>MIR6859-2</td>\n",
       "      <td>unk</td>\n",
       "      <td>unk</td>\n",
       "      <td>-1,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   number Name of Gene Chromosome Strand      Start        End  \\\n",
       "0     585    NR_148357       chr1      +      11868      14362   \n",
       "1     585    NR_046018       chr1      +      11873      14409   \n",
       "2    2127    NM_139163       chr2      -  202152993  202222121   \n",
       "3     585    NR_106918       chr1      -      17368      17436   \n",
       "4     585    NR_107062       chr1      -      17368      17436   \n",
       "\n",
       "   Coding region start  Coding region end  Number of exons  \\\n",
       "0                14362              14362                3   \n",
       "1                14409              14409                3   \n",
       "2            202153377          202216127               15   \n",
       "3                17436              17436                1   \n",
       "4                17436              17436                1   \n",
       "\n",
       "                                 Exon start position  \\\n",
       "0                                 11868,12612,13220,   \n",
       "1                                 11873,12612,13220,   \n",
       "2  202152993,202154179,202154418,202163960,202172...   \n",
       "3                                             17368,   \n",
       "4                                             17368,   \n",
       "\n",
       "                                   Exon end position  Score Alternative name  \\\n",
       "0                                 12227,12721,14362,      0     LOC102725121   \n",
       "1                                 12227,12721,14409,      0          DDX11L1   \n",
       "2  202153504,202154289,202154577,202164023,202172...      0           FLACC1   \n",
       "3                                             17436,      0        MIR6859-1   \n",
       "4                                             17436,      0        MIR6859-2   \n",
       "\n",
       "  Status of CDS start annotations(none, unnk, incmpl, cmpl)  \\\n",
       "0                                                unk          \n",
       "1                                                unk          \n",
       "2                                               cmpl          \n",
       "3                                                unk          \n",
       "4                                                unk          \n",
       "\n",
       "  Status of CDS end annotations(none, unnk, incmpl, cmpl)  \\\n",
       "0                                                unk        \n",
       "1                                                unk        \n",
       "2                                               cmpl        \n",
       "3                                                unk        \n",
       "4                                                unk        \n",
       "\n",
       "                exon frame offsets  \n",
       "0                        -1,-1,-1,  \n",
       "1                        -1,-1,-1,  \n",
       "2  2,0,0,0,1,0,1,2,0,2,0,2,2,0,-1,  \n",
       "3                              -1,  \n",
       "4                              -1,  "
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refGene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "os.chdir('/data/share/ICGC/STAD/all_snv_indel/fix_all_snv_indel_190220/')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['com.indel',\n",
       " 'Multi-center_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rmSNP.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'China_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'com.mv',\n",
       " 'China-WGS-target0_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'RCAST_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rmSNP.rm-new_f0002.add-paired-NVAF',\n",
       " 'Singapore_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rmSNP.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'NCC_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF',\n",
       " 'HongKong-WGS-target0_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'HongKong-WGS-target0_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'China-WGS-target0_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Singapore_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'RCAST_hyper_all_snv_new2.txt.rmSNP.rm-new_f0002.add-paired-NVAF.1',\n",
       " 'TCGA_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'HongKong-WGS-target0_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'HongKong-WGS-target0_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'RCAST_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF',\n",
       " 'China_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Korea_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'Korea_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Multi-center_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'NCC_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.1',\n",
       " 'NCC_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.1',\n",
       " 'Singapore_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Korea_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'RCAST_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.1',\n",
       " 'Korea_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Multi-center_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'TCGA_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'China_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'China-WGS-target0_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'China-WGS-target0_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'Multi-center_hyper_all_snv_new2.txt.rmSNP.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'com.snv',\n",
       " 'NCC_hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF',\n",
       " 'TCGA_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'China_non-hyper_all_indel_new2.txt.rm_in-frame-new_n1_f0001.rm-new_f0002.add-paired-NVAF.chName',\n",
       " 'TCGA_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Singapore_hyper_all_snv_new2.txt.rmSNP.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'com.mv.1']"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "os.listdir()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['HongKong-WGS-target0_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'HongKong-WGS-target0_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'China-WGS-target0_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'RCAST_hyper_all_snv_new2.txt.rmSNP.rm-new_f0002.add-paired-NVAF.1',\n",
       " 'China_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Korea_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'NCC_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.1',\n",
       " 'NCC_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.1',\n",
       " 'Singapore_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'RCAST_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.1',\n",
       " 'Korea_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Multi-center_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'China_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'China-WGS-target0_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Multi-center_hyper_all_snv_new2.txt.rmSNP.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'TCGA_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'TCGA_hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.chName.1',\n",
       " 'Singapore_hyper_all_snv_new2.txt.rmSNP.rm-new_f0002.add-paired-NVAF.chName.1']"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[item for item in os.listdir() if \"all_snv\" in item]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "snv1 = pd.read_csv(\"NCC_non-hyper_all_snv_new2.txt.rm-new_f0002.add-paired-NVAF.1\", header=None, sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>12</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>17</th>\n",
       "      <th>18</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>21</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>26610773</td>\n",
       "      <td>26610773</td>\n",
       "      <td>93</td>\n",
       "      <td>51129</td>\n",
       "      <td>0.161290</td>\n",
       "      <td>0.000254</td>\n",
       "      <td>1353</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15</td>\n",
       "      <td>635.000000</td>\n",
       "      <td>-777</td>\n",
       "      <td>-777</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>87</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>32051123</td>\n",
       "      <td>32051123</td>\n",
       "      <td>73</td>\n",
       "      <td>35900</td>\n",
       "      <td>0.369863</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>1353</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "      <td>1896.733333</td>\n",
       "      <td>-777</td>\n",
       "      <td>-777</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>87</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>69498927</td>\n",
       "      <td>69498927</td>\n",
       "      <td>291</td>\n",
       "      <td>98693</td>\n",
       "      <td>0.202749</td>\n",
       "      <td>0.000203</td>\n",
       "      <td>676</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>59</td>\n",
       "      <td>998.763547</td>\n",
       "      <td>-777</td>\n",
       "      <td>-777</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>252</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>90470875</td>\n",
       "      <td>90470875</td>\n",
       "      <td>67</td>\n",
       "      <td>22487</td>\n",
       "      <td>0.417910</td>\n",
       "      <td>0.000133</td>\n",
       "      <td>1323</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "      <td>3142.180451</td>\n",
       "      <td>-777</td>\n",
       "      <td>-777</td>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>40</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>94643587</td>\n",
       "      <td>94643587</td>\n",
       "      <td>234</td>\n",
       "      <td>73670</td>\n",
       "      <td>0.260684</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>1353</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>61</td>\n",
       "      <td>835.525641</td>\n",
       "      <td>A/G</td>\n",
       "      <td>+</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>190</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       0   1   2         3         4    5      6         7         8     9   \\\n",
       "0  AFP01T   1   1  26610773  26610773   93  51129  0.161290  0.000254  1353   \n",
       "1  AFP01T   1   1  32051123  32051123   73  35900  0.369863  0.000195  1353   \n",
       "2  AFP01T   1   1  69498927  69498927  291  98693  0.202749  0.000203   676   \n",
       "3  AFP01T   1   1  90470875  90470875   67  22487  0.417910  0.000133  1323   \n",
       "4  AFP01T   1   1  94643587  94643587  234  73670  0.260684  0.000312  1353   \n",
       "\n",
       "   ...  12   13  14           15    16    17   18            19   20   21  \n",
       "0  ...   0  0.0  15   635.000000  -777  -777  C>T  2.200000e-16   87  0.0  \n",
       "1  ...   0  0.0  27  1896.733333  -777  -777  T>G  2.200000e-16   87  0.0  \n",
       "2  ...   0  0.0  59   998.763547  -777  -777  A>G  2.200000e-16  252  0.0  \n",
       "3  ...   0  0.0  28  3142.180451  -777  -777  T>A  2.200000e-16   40  0.0  \n",
       "4  ...   0  0.0  61   835.525641   A/G     +  G>A  2.200000e-16  190  0.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snv1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "colnames = [\"Sample\", \"type (1:SNV, 2:Ins, 3: Del)\", \"Chromosome\",\"Start\", \"End\",\"Tumor depth\", \"normal depth\",\n",
    "           \"TVAF\",\"NVAF\",\"10\",\"11\",\"12\",\"13\",\"14\",\"Number of read of mutation\", \"TVAF/NVAF\", \"dbSNP\", \"dnSNP strand\", \"mutation\", \"p-value\", \n",
    "           \"paired normal depth\", \"paired normal VAF\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "22"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(colnames)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "snv1.columns = colnames"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1+1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>number</th>\n",
       "      <th>Name of Gene</th>\n",
       "      <th>Chromosome name</th>\n",
       "      <th>'+' or '-' for strand</th>\n",
       "      <th>Transcription start position</th>\n",
       "      <th>Transcription end position</th>\n",
       "      <th>Coding region start</th>\n",
       "      <th>Coding region end</th>\n",
       "      <th>Number of exons</th>\n",
       "      <th>Exon start position</th>\n",
       "      <th>Exon end position</th>\n",
       "      <th>Score</th>\n",
       "      <th>Alternative name</th>\n",
       "      <th>Status of CDS start annotations(none, unnk, incmpl, cmpl)</th>\n",
       "      <th>Status of CDS end annotations(none, unnk, incmpl, cmpl)</th>\n",
       "      <th>exon frame offsets</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>585</td>\n",
       "      <td>NR_148357</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>11868</td>\n",
       "      <td>14362</td>\n",
       "      <td>14362</td>\n",
       "      <td>14362</td>\n",
       "      <td>3</td>\n",
       "      <td>11868,12612,13220,</td>\n",
       "      <td>12227,12721,14362,</td>\n",
       "      <td>0</td>\n",
       "      <td>LOC102725121</td>\n",
       "      <td>unk</td>\n",
       "      <td>unk</td>\n",
       "      <td>-1,-1,-1,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>585</td>\n",
       "      <td>NR_046018</td>\n",
       "      <td>chr1</td>\n",
       "      <td>+</td>\n",
       "      <td>11873</td>\n",
       "      <td>14409</td>\n",
       "      <td>14409</td>\n",
       "      <td>14409</td>\n",
       "      <td>3</td>\n",
       "      <td>11873,12612,13220,</td>\n",
       "      <td>12227,12721,14409,</td>\n",
       "      <td>0</td>\n",
       "      <td>DDX11L1</td>\n",
       "      <td>unk</td>\n",
       "      <td>unk</td>\n",
       "      <td>-1,-1,-1,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2127</td>\n",
       "      <td>NM_139163</td>\n",
       "      <td>chr2</td>\n",
       "      <td>-</td>\n",
       "      <td>202152993</td>\n",
       "      <td>202222121</td>\n",
       "      <td>202153377</td>\n",
       "      <td>202216127</td>\n",
       "      <td>15</td>\n",
       "      <td>202152993,202154179,202154418,202163960,202172...</td>\n",
       "      <td>202153504,202154289,202154577,202164023,202172...</td>\n",
       "      <td>0</td>\n",
       "      <td>FLACC1</td>\n",
       "      <td>cmpl</td>\n",
       "      <td>cmpl</td>\n",
       "      <td>2,0,0,0,1,0,1,2,0,2,0,2,2,0,-1,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>585</td>\n",
       "      <td>NR_106918</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>17368</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>1</td>\n",
       "      <td>17368,</td>\n",
       "      <td>17436,</td>\n",
       "      <td>0</td>\n",
       "      <td>MIR6859-1</td>\n",
       "      <td>unk</td>\n",
       "      <td>unk</td>\n",
       "      <td>-1,</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>585</td>\n",
       "      <td>NR_107062</td>\n",
       "      <td>chr1</td>\n",
       "      <td>-</td>\n",
       "      <td>17368</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>1</td>\n",
       "      <td>17368,</td>\n",
       "      <td>17436,</td>\n",
       "      <td>0</td>\n",
       "      <td>MIR6859-2</td>\n",
       "      <td>unk</td>\n",
       "      <td>unk</td>\n",
       "      <td>-1,</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   number Name of Gene Chromosome name '+' or '-' for strand  \\\n",
       "0     585    NR_148357            chr1                     +   \n",
       "1     585    NR_046018            chr1                     +   \n",
       "2    2127    NM_139163            chr2                     -   \n",
       "3     585    NR_106918            chr1                     -   \n",
       "4     585    NR_107062            chr1                     -   \n",
       "\n",
       "   Transcription start position  Transcription end position  \\\n",
       "0                         11868                       14362   \n",
       "1                         11873                       14409   \n",
       "2                     202152993                   202222121   \n",
       "3                         17368                       17436   \n",
       "4                         17368                       17436   \n",
       "\n",
       "   Coding region start  Coding region end  Number of exons  \\\n",
       "0                14362              14362                3   \n",
       "1                14409              14409                3   \n",
       "2            202153377          202216127               15   \n",
       "3                17436              17436                1   \n",
       "4                17436              17436                1   \n",
       "\n",
       "                                 Exon start position  \\\n",
       "0                                 11868,12612,13220,   \n",
       "1                                 11873,12612,13220,   \n",
       "2  202152993,202154179,202154418,202163960,202172...   \n",
       "3                                             17368,   \n",
       "4                                             17368,   \n",
       "\n",
       "                                   Exon end position  Score Alternative name  \\\n",
       "0                                 12227,12721,14362,      0     LOC102725121   \n",
       "1                                 12227,12721,14409,      0          DDX11L1   \n",
       "2  202153504,202154289,202154577,202164023,202172...      0           FLACC1   \n",
       "3                                             17436,      0        MIR6859-1   \n",
       "4                                             17436,      0        MIR6859-2   \n",
       "\n",
       "  Status of CDS start annotations(none, unnk, incmpl, cmpl)  \\\n",
       "0                                                unk          \n",
       "1                                                unk          \n",
       "2                                               cmpl          \n",
       "3                                                unk          \n",
       "4                                                unk          \n",
       "\n",
       "  Status of CDS end annotations(none, unnk, incmpl, cmpl)  \\\n",
       "0                                                unk        \n",
       "1                                                unk        \n",
       "2                                               cmpl        \n",
       "3                                                unk        \n",
       "4                                                unk        \n",
       "\n",
       "                exon frame offsets  \n",
       "0                        -1,-1,-1,  \n",
       "1                        -1,-1,-1,  \n",
       "2  2,0,0,0,1,0,1,2,0,2,0,2,2,0,-1,  \n",
       "3                              -1,  \n",
       "4                              -1,  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "refGene.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample</th>\n",
       "      <th>type (1:SNV, 2:Ins, 3: Del)</th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Start</th>\n",
       "      <th>End</th>\n",
       "      <th>Tumor depth</th>\n",
       "      <th>normal depth</th>\n",
       "      <th>TVAF</th>\n",
       "      <th>NVAF</th>\n",
       "      <th>10</th>\n",
       "      <th>...</th>\n",
       "      <th>13</th>\n",
       "      <th>14</th>\n",
       "      <th>Number of read of mutation</th>\n",
       "      <th>TVAF/NVAF</th>\n",
       "      <th>dbSNP</th>\n",
       "      <th>dnSNP strand</th>\n",
       "      <th>mutation</th>\n",
       "      <th>p-value</th>\n",
       "      <th>paired normal depth</th>\n",
       "      <th>paired normal VAF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>26610773</td>\n",
       "      <td>26610773</td>\n",
       "      <td>93</td>\n",
       "      <td>51129</td>\n",
       "      <td>0.161290</td>\n",
       "      <td>0.000254</td>\n",
       "      <td>1353</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15</td>\n",
       "      <td>635.000000</td>\n",
       "      <td>-777</td>\n",
       "      <td>-777</td>\n",
       "      <td>C&gt;T</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>87</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>32051123</td>\n",
       "      <td>32051123</td>\n",
       "      <td>73</td>\n",
       "      <td>35900</td>\n",
       "      <td>0.369863</td>\n",
       "      <td>0.000195</td>\n",
       "      <td>1353</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>27</td>\n",
       "      <td>1896.733333</td>\n",
       "      <td>-777</td>\n",
       "      <td>-777</td>\n",
       "      <td>T&gt;G</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>87</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>69498927</td>\n",
       "      <td>69498927</td>\n",
       "      <td>291</td>\n",
       "      <td>98693</td>\n",
       "      <td>0.202749</td>\n",
       "      <td>0.000203</td>\n",
       "      <td>676</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>59</td>\n",
       "      <td>998.763547</td>\n",
       "      <td>-777</td>\n",
       "      <td>-777</td>\n",
       "      <td>A&gt;G</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>252</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>90470875</td>\n",
       "      <td>90470875</td>\n",
       "      <td>67</td>\n",
       "      <td>22487</td>\n",
       "      <td>0.417910</td>\n",
       "      <td>0.000133</td>\n",
       "      <td>1323</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>28</td>\n",
       "      <td>3142.180451</td>\n",
       "      <td>-777</td>\n",
       "      <td>-777</td>\n",
       "      <td>T&gt;A</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>40</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AFP01T</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>94643587</td>\n",
       "      <td>94643587</td>\n",
       "      <td>234</td>\n",
       "      <td>73670</td>\n",
       "      <td>0.260684</td>\n",
       "      <td>0.000312</td>\n",
       "      <td>1353</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>61</td>\n",
       "      <td>835.525641</td>\n",
       "      <td>A/G</td>\n",
       "      <td>+</td>\n",
       "      <td>G&gt;A</td>\n",
       "      <td>2.200000e-16</td>\n",
       "      <td>190</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 22 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   Sample  type (1:SNV, 2:Ins, 3: Del)  Chrom     Start       End  \\\n",
       "0  AFP01T                            1      1  26610773  26610773   \n",
       "1  AFP01T                            1      1  32051123  32051123   \n",
       "2  AFP01T                            1      1  69498927  69498927   \n",
       "3  AFP01T                            1      1  90470875  90470875   \n",
       "4  AFP01T                            1      1  94643587  94643587   \n",
       "\n",
       "   Tumor depth  normal depth      TVAF      NVAF    10  ...  13   14  \\\n",
       "0           93         51129  0.161290  0.000254  1353  ...   0  0.0   \n",
       "1           73         35900  0.369863  0.000195  1353  ...   0  0.0   \n",
       "2          291         98693  0.202749  0.000203   676  ...   0  0.0   \n",
       "3           67         22487  0.417910  0.000133  1323  ...   0  0.0   \n",
       "4          234         73670  0.260684  0.000312  1353  ...   0  0.0   \n",
       "\n",
       "   Number of read of mutation    TVAF/NVAF  dbSNP  dnSNP strand mutation  \\\n",
       "0                          15   635.000000   -777          -777      C>T   \n",
       "1                          27  1896.733333   -777          -777      T>G   \n",
       "2                          59   998.763547   -777          -777      A>G   \n",
       "3                          28  3142.180451   -777          -777      T>A   \n",
       "4                          61   835.525641    A/G             +      G>A   \n",
       "\n",
       "        p-value paired normal depth  paired normal VAF  \n",
       "0  2.200000e-16                  87                0.0  \n",
       "1  2.200000e-16                  87                0.0  \n",
       "2  2.200000e-16                 252                0.0  \n",
       "3  2.200000e-16                  40                0.0  \n",
       "4  2.200000e-16                 190                0.0  \n",
       "\n",
       "[5 rows x 22 columns]"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snv1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(83900, 22)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snv1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "T>G -> A>C"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "gr1 = PyRanges(refGene)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "+-----------+----------------+--------------+--------------+-----------+-----------+-----------------------+---------------------+-------------------+-------+\n",
       "| number    | Name of Gene   | Chromosome   | Strand       | Start     | End       | Coding region start   | Coding region end   | Number of exons   | ...   |\n",
       "| (int64)   | (object)       | (category)   | (category)   | (int32)   | (int32)   | (int64)               | (int64)             | (int64)           | ...   |\n",
       "|-----------+----------------+--------------+--------------+-----------+-----------+-----------------------+---------------------+-------------------+-------|\n",
       "| 585       | NR_148357      | chr1         | +            | 11868     | 14362     | 14362                 | 14362               | 3                 | ...   |\n",
       "| 585       | NR_046018      | chr1         | +            | 11873     | 14409     | 14409                 | 14409               | 3                 | ...   |\n",
       "| 585       | NR_036051      | chr1         | +            | 30365     | 30503     | 30503                 | 30503               | 1                 | ...   |\n",
       "| 585       | NR_036266      | chr1         | +            | 30365     | 30503     | 30503                 | 30503               | 1                 | ...   |\n",
       "| ...       | ...            | ...          | ...          | ...       | ...       | ...                   | ...                 | ...               | ...   |\n",
       "| 768       | NM_001006121   | chrY         | -            | 24026222  | 24040673  | 24026500              | 24038660            | 12                | ...   |\n",
       "| 768       | NM_005058      | chrY         | -            | 24026222  | 24040673  | 24026500              | 24038660            | 12                | ...   |\n",
       "| 768       | NM_001006121   | chrY         | -            | 24049764  | 24064214  | 24050042              | 24062201            | 12                | ...   |\n",
       "| 792       | NM_004678      | chrY         | -            | 27177048  | 27198251  | 27184244              | 27190170            | 9                 | ...   |\n",
       "+-----------+----------------+--------------+--------------+-----------+-----------+-----------------------+---------------------+-------------------+-------+\n",
       "Stranded PyRanges object has 74,873 rows and 16 columns from 54 chromosomes.\n",
       "Hidden columns: Exon start position, Exon end position, Score, Alternative name, Status of CDS start annotations(none, unnk, incmpl, cmpl), Status of CDS end annotations(none, unnk, incmpl, cmpl), exon frame offsets"
      ]
     },
     "execution_count": 79,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gr1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
